cmd/juju/application/bundle.go

```diff
@@ -1758,3 +1780,11 @@ func applicationRequiresTrust(appSpec *charm.ApplicationSpec) bool {
 	optRequiresTrust := appSpec.Options != nil && appSpec.Options["trust"] == true
 	return appSpec.RequiresTrust || optRequiresTrust
 }
+
+// isUserAlreadyHasAccessErr returns true if err indicates that the user
+// already has access to an offer. Unfortunately, the server does not set a
+// status code for this error so we need to fall back to a hacky string
+// comparison to be compatible with older controllers.
+func isUserAlreadyHasAccessErr(err error) bool {
```

Comment:
Can we have a constant for the error string?

Also can we rename the function as it reads better.
```suggestion
func hasUserAlreadyHaveAccessErr(err error) bool {
```

Reply:
TIL "suggestion" markdown.